packages/stockflux-watchlist/src/components/watchlist/Watchlist.js

```diff
@@ -114,18 +122,81 @@ const Watchlist = () => {
     setWatchlist(watchlist.filter(item => item !== symbol));
   };
 
+  const getWindowOptions = async () => {
+    const targetApplication = await OpenfinApiHelpers.getStockFluxApp(
+      'stockflux-chart'
+    );
+    const manifestContents = await fetch(targetApplication.manifest, {
+      method: 'GET'
+    });
+
+    const info = await manifestContents.json();
+    return info.startup_app;
+  };
+
+  useEffect(() => {
+    getWindowOptions().then(value => {
+      setPreviewOptions(value);
+    });
+  }, []);
+
+  useEffect(() => {
+    const windowOffset = 5;
+    const screenLeft =
+      window.screenLeft > 0
+        ? window.screenLeft
+        : window.screen.availWidth + window.screenLeft;
+    if (previewOptions) {
+      const leftPosition =
+        window.screen.availWidth -
+          (window.outerWidth +
+            screenLeft +
+            windowOffset +
+            previewOptions.defaultWidth) >
+        0
+          ? window.outerWidth + screenLeft + windowOffset
+          : screenLeft - windowOffset - previewOptions.defaultWidth;
+      setPreviewPosition({
+        left:
+          leftPosition > window.screen.availLeft
+            ? window.screenLeft > 0
+              ? leftPosition
+              : window.screen.availLeft + leftPosition
+            : window.outerWidth + screenLeft + windowOffset,
```

Comment:
Is there any way of using a fixed position to reduce the amount of ternary statements needed?

Reply:
I have tried to avoid using fixed positions, as the watch list window can be moved around, and i would prefer the popup located to be around that. The position is only calculated when the box is to be shown, as this handled is the parent window has been moved.  
The reason there is so many checks on positions as it should handle multiple monitors correctly as well with showing the box on the side which has the most space to fit it within without overlapping the watch list box itself.

Reply:
Maybe we should split these into small helper functions, aka each ternary should be its own function? I think it might make it more readable.

Reply:
Refactored the code to make the function more readable.